api/v1beta1/common.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/endpoint"
+)
+
+// GlanceAPITemplate defines the desired state of GlanceAPI
+type GlanceAPITemplate struct {
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=glance
+	// ServiceUser - optional username used for this service to register in glance
+	ServiceUser string `json:"serviceUser"`
+
+	// +kubebuilder:validation:Required
+	// GlanceAPI Container Image URL
+	ContainerImage string `json:"containerImage"`
```

Comment:
It might worth to thing through if we need to set this differently per GlanceAPIInternalSpec / GlanceAPIExternalSpec. If we need to set it differently then yes, we need this in the Template, but if the two always have the same value then we can remove ContainerImage from the Template and copy the value from Glance.Spec.ContainerImage when creating the GlanceAPISpec struct.

Reply:
Interesting, I think it's not supported having different images for External/Internal (and with webhooks we always enforce the same image), so I guess it makes sense removing this part from the template

Reply:
Actually, I'm thinking about keeping this parameter in the `Template` because I'm referencing the struct in [1] to inject the images via webhooks: this would keep the current logic unchanged.

[1] https://github.com/openstack-k8s-operators/glance-operator/blob/master/api/v1beta1/glance_webhook.go#L66

Reply:
I think the code in [1] is just a consequence of having two separate levels where we can set the ContainerImage. I.e. if we have two levels to specify it then we need two levels to default it. But I'm OK to keep ContainerImage now. We can decide if we need two levels of specifying it later.